src/storage/patch/RdfImmutableCheckPatcher.ts

```diff
@@ -0,0 +1,75 @@
+import type { Quad } from 'rdf-js';
+import type { AuxiliaryStrategy } from '../../http/auxiliary/AuxiliaryStrategy';
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { ConflictHttpError } from '../../util/errors/ConflictHttpError';
+import { serializeQuads, uniqueQuads } from '../../util/QuadUtil';
+import { cloneRepresentation } from '../../util/ResourceUtil';
+import { readableToQuads, readableToString } from '../../util/StreamUtil';
+import { RDF, PIM, LDP } from '../../util/Vocabularies';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+export class RdfImmutableCheckPatcher extends RepresentationPatcher {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly patcher: RepresentationPatcher;
+  private readonly metaStrategy: AuxiliaryStrategy;
+
+  public constructor(patcher: RepresentationPatcher, metaStrategy: AuxiliaryStrategy) {
+    super();
+    this.patcher = patcher;
+    this.metaStrategy = metaStrategy;
+  }
+
+  public async handle(input: RepresentationPatcherInput): Promise<Representation> {
+    let { representation } = input;
+    const { identifier } = input;
+
+    representation = representation ?? new BasicRepresentation([], INTERNAL_QUADS);
+    const immutableTriples: [string, string | undefined][] = [];
+    immutableTriples.push([ RDF.type, PIM.Storage ]);
+    immutableTriples.push([ LDP.contains, undefined ]);
```

Comment:
This list should be provided as a constructor parameter through Components.js so it can easily be modified without having to change the code.

Besides that the list is not complete, this list from the spec needs to be included: https://solidproject.org/TR/protocol#contained-resource-metadata-statements.

Reply:
Just realized that I also interpreted the list in the link wrong. This is the metadata of the resources contained in the container, which you see when doing a GET on a container, not metadata of the container itself. But most of them are still good examples of triples that should not be edited.